File: 2249-count-the-hidden-sequences/count-the-hidden-sequences.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        total = 0
        min_prefix, max_prefix = 0, 0

        for diff in differences:
            total += diff
            min_prefix = min(min_prefix, total)
            max_prefix = max(max_prefix, total)

        min_start = lower - min_prefix
        max_start = upper - max_prefix

        return max(0, max_start - min_start + 1)

File: 2249-count-the-hidden-sequences/count-the-hidden-sequences.js
var numberOfArrays = function(differences, lower, upper) {
    let minPrefix = 0, maxPrefix = 0, sum = 0;

    for (let diff of differences) {
        sum += diff;
        minPrefix = Math.min(minPrefix, sum);
        maxPrefix = Math.max(maxPrefix, sum);
    }

    const minStart = lower - minPrefix;
    const maxStart = upper - maxPrefix;

    return Math.max(0, maxStart - minStart + 1);
};